peps/pep-0788.rst

```diff
@@ -0,0 +1,633 @@
+PEP: 788
+Title: Reimagining native threads
+Author: Peter Bierma <zintensitydev@gmail.com>
+Sponsor: Victor Stinner <vstinner@python.org>
+Discussions-To: Pending
+Status: Draft
+Type: Standards Track
+Created: 23-Apr-2025
+Python-Version: 3.15
+Post-History: `10-Mar-2025 <https://discuss.python.org/t/83959>`__
+
+
+Abstract
+========
+
+:c:func:`PyGILState_Ensure`, :c:func:`PyGILState_Ensure`, and other related
+functions in the ``PyGILState`` family are the most common way to create
+native threads that interact with Python. They have been the standard for over
+twenty years (:pep:`311`). But, over time, these functions have
+become problematic:
+
+- They aren't safe for finalization, either causing the calling thread to hang or
+  crashing it with a segmentation fault, preventing further execution.
+- When they're called before finalization, they force the thread to be
+  "daemon", meaning that the interpreter won't wait for it to reach any point
+  of execution. This is mostly frustrating for developers, but can lead to
+  deadlocks!
+- Subinterpreters don't play nicely with them, because they all assume that
+  the main interpreter is the only one that exists. A fresh thread (*i.e.*,
+  has never had a thread state) that calls :c:func:`PyGILState_Ensure` will
+  always be for the main interpreter.
+- The term "GIL" in the name is quite confusing for users of free-threaded
+  Python. There isn't a GIL, why do they still have to call it?
+
+This PEP intends to fix all of these issues by providing two new functions,
+:c:func:`PyThreadState_Ensure` and :c:func:`PyThreadState_Release`, as a more
+correct and safer replacement for :c:func:`PyGILState_Ensure` and
+:c:func:`PyGILState_Release`. For example:
+
+.. code-block:: C
```

Comment:
Usually lower case is used:
```suggestion
.. code-block:: c
```
Or we can put `.. highlight:: c` near the top to set the default, and remove these, and change `For example:` to `For example::`